fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -298,7 +298,7 @@ def __init__(
         self,
         module: nn.Module,
         process_group: Optional[ProcessGroup] = None,
-        process_group_reduce_scatter: Union[ProcessGroup, ProcessGroupName] = ProcessGroupName.reduce_scatter,
+        process_group_reduce_scatter: Any = ProcessGroupName.reduce_scatter,
```

Comment:
It has to be Any. Or else it would not pass mypy type check with the error message. 
" Incompatible types in assignment (expression has type "ProcessGroupName",
variable has type "ProcessGroup")
    ...             self.process_group_reduce_scatter = process_group_reduce_... "
I added extra type check in the code below to ensure correct type is passed in.

Reply:
Can you add a comment here please?